### DISCUSSION TITLE
Relative paths in meta tags

### DISCUSSION THREAD

**Discussion by djr319** [webpack/webpack#16385] [Q&A]
Hi, I'm new to Webpack 

I'm building a multi-page static website.

My webpack5 built website has relative links ("../") to assets, including  within 
`<meta property="og:image" content="../assets/images/picture.jpg" />`. (statically written in the HTML file)

Behaviour:
- Links to assets (injected with JS) within the page `<body>` are resolved to absolute paths on build ✅ 
- Paths to assets in the meta tag remain relative (and do not find the resource because of my changed file structure on build) ❌

I've tried writing the relative path in both the HTML and injecting with JS (because I' not too sure how it works exactly!). Looks like I need to hard code the absolute paths.

Thanks for any pointers.. See my config below (and feel free to offer suggestions if there's any other feedback)
<details>
<summary>webpack.config.js</summary>

```js
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const htmlPageNames = ['...'];

module.exports = {
  mode: 'production',
  entry: htmlPageNames.reduce((config, page) => {
    config[page] = `./src/js/${page}.js`;
    return config;
  }, {
    scss: `./src/js/scss.js`,
  }),
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].[contenthash].js',
    clean: true,
    assetModuleFilename: 'assets/[name][ext]'
  },
  plugins: [].concat(
    htmlPageNames.map(
      (page) =>
        new HtmlWebpackPlugin({
          inject: true,
          template: `./src/html/${page}.html`,
          filename: `${page}.html`,
          chunks: [page],
        })
    )
  ),

  module: {
    rules: [
      {
        test: /\.(png|svg|jpg|jpeg|gif)$/i,
        type: 'asset',
      },
      {
        test: /\.(scss)$/,
        use: [
          'style-loader', // Creates `style` nodes from JS strings
          'css-loader', // Translates CSS into CommonJS
          {
            loader: 'postcss-loader',
            options: {
              postcssOptions: {
                plugins: () => [
                  require('autoprefixer')
                ]
              }
            }
          },
          {
            loader: 'sass-loader' // Compiles Sass to CSS
          }
        ]
      }
    ]
  },
  optimization: {
    splitChunks: {
      chunks: "all",
    },
  },
  performance: {
    hints: false,
  },
  devServer: {
    static: {
      directory: path.resolve(__dirname, 'dist')
    },
    port: 8080,
    open: true,
    hot: true,
    compress: true,
    historyApiFallback: true,
  }
};
```

</details>

**Comment by alexander-akait** [comment]
Where you generate `<meta property="og:image" content="../assets/images/picture.jpg" />`?

**Reply by djr319** [reply]
Hi Alex, I've tried both statically hard coded in the html, and injecting in the associated .js bundle for that page.

**Reply by alexander-akait** [reply]
You can have `copy-webpack-plugin` (i.e. copy tham in dist/public) and put tags with link on them, but if you have different image on different pages, you need a plugin or something else to generate them dynamically

**Reply by djr319** [reply]
Ah yes, I'm using `copy-webpack-plugin` for my favicon, I suppose that would be an option as well. Thanks for your help and quick replies Alexander 🙏
